### ISSUE TITLE
Re-using scope after flushing

### ISSUE THREAD

**Issue by jstans** [SpoonX/wetland#243]
Is this doable? a second flush doesn't seem to update the state (or throw an error), even if doing a .refresh() between. I appreciate that this may not be the original idea behind the scope, but there are certainly instances where you want to write out current changes to the database for consistency without having to create a new scope. Especially in making sure another process reading from the database doesn't trigger on old data.

Might be something simple I have missed but it's not something I could find in the documentation and from an intuitive point of view I simply assumed this would work, in addition to updating the id in the scope (though this would presumably need returning support in the dialect or triggering another select).

**Comment by RWOverdijk** [comment]
After flushing, you _can_ reuse a scope. That's the way it was designed. To be a full-fledged EntityManager that's disposable between requests.

What I'm assuming, is that you're persisting new entities, and then proceed to make changes to them, expecting them to show up as **dirty** (modified). If that's the case, keep reading! :)

This won't work. Entities fetched from the database using the Scope, will be observed by default. New entities won't. New entity instances aren't wrapped (proxied) yet, they aren't being observed for changes. This is a performance optimisation, given that you generally don't modify a persisted entity in the same context.

### TL;DR;

What you can do, is the following. You "attach" the entity, making sure it's observed.

```js
const user = new User;

scope
  .attach(user)
  .persist(user)
  .flush()
  .then(() => {
    // Go nuts! The entity is observed anyway.

    user.lastName = 'Polo';

    return scope.flush();
  });
```

Now, the entity is observed for changes, and being persisted. After flushing, you can continue working with the same reference.

I hope this helps!

**Comment by jstans** [comment]
Perfect, thanks. Though this could definitely do with something added to the scope tutorial.

**Comment by RWOverdijk** [comment]
I agree. The docs could use some work in general!
